### ISSUE TITLE
服务器发生同步聊天时会导致主线程休眠

### ISSUE THREAD

**Issue by BryanSer** [lintx/Minecraft-Plugin-YinwuChat#2]
https://github.com/lintx/Minecraft-Plugin-YinwuChat/blob/4b0144a616ffb5f4d730222d2d9e08cbaac00b31/src/main/java/org/lintx/plugins/yinwuchat/bukkit/Listeners.java#L23

如果你要问 什么情况会发生同步聊天
那建议你看一眼Player的chat方法的实现 代码如下
```java
  public void chat(String msg) {
    if ((getHandle()).playerConnection == null)
      return; 
    (getHandle()).playerConnection.chat(msg, false);
  }
```
其中playerConnection的第二个参数为async
如果你不对同步进行判断的话
如果有其他插件调用了聊天方法则会导致主线程被你休眠
请加入判断
`event.isAsynchronous()`

**Comment by lintx** [comment]
谢谢，将在下版本修复，同步聊天可能是老版本的方法，较新的版本都是异步的了，然后由此会引出一些需要监听消息来实现功能的部分插件出现冲突所以才加的这个可选的配置，不过看来还是没有考虑周全

**Comment by BryanSer** [comment]
刚刚节选的代码是1.13的spigot的craftbukkit实现
刚刚在审核你的优秀 顺手看到了

**Comment by lintx** [comment]
我看了1.13.2的craftbukkit，但是因为同步聊天已经被标记为过期，所以没有过多关注同步聊天的，下版本会修复，并将默认等待时间修改为0，谢谢了

**Comment by lintx** [comment]
这个还是这版本新增的/捂脸，因为有人反应有一个仓库插件存经验和钱存不进去/捂脸

**Comment by BryanSer** [comment]
不过我需要说明一点
被标记为过时的事件是PlayerChatEvent 这个在很早很早的版本就抛弃了
而我刚刚说的
即使是AsyncPlayerChatEvent 也会发生同步情况

**Comment by BryanSer** [comment]
在文档中也有说明

```
This event will sometimes fire synchronously, depending on how it was triggered.
The constructor provides a boolean to indicate if the event was fired synchronously or asynchronously. When asynchronous, this event can be called from any thread, sans the main thread, and has limited access to the API.

If a player is the direct cause of this event by an incoming packet, this event will be asynchronous. If a plugin triggers this event by compelling a player to chat, this event will be synchronous.

Care should be taken to check Event.isAsynchronous() and treat the event appropriately.
```

**Comment by lintx** [comment]
好的，预计今晚或明天会发布下个版本，下个版本预计还会修复另一个bungee插件消息被取消导致其他插件无法读取的bug和增加子服私聊的自动填写其他子服玩家名的功能

**Comment by lintx** [comment]
已修复
